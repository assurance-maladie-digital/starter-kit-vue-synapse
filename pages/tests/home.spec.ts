import { mount } from '@vue/test-utils'
import Home from '../home.vue'
import { describe, it, expect } from 'vitest'
import { vuetify } from '@/tests/unit/setup'

describe('home page should render', () => {
	it('should render', async () => {
		const wrapper = mount(Home, {
			global: {
				plugins: [vuetify],
			},
		})

		expect(wrapper.html()).toMatchSnapshot()

		const h1Text = wrapper.find('h1').text()
		expect(h1Text).toContain('Accueil')
	})

	it('renders title from $config', async () => {
		const mockConfig = {
			public: {
				title: 'Test Title',
			},
		}

		const wrapper = mount(Home, {
			global: {
				plugins: [vuetify],
				mocks: {
					$config: mockConfig,
				},
			},
		})

		const h1Text = wrapper.find('h1').text()
		expect(h1Text).toContain('Test Title')
	})

	it('renders default title when $config is not available', async () => {
		const wrapper = mount(Home, {
			global: {
				plugins: [vuetify],
				mocks: {
					$config: null,
				},
			},
		})

		const h1Text = wrapper.find('h1').text()
		expect(h1Text).toContain('Accueil')
	})
})
