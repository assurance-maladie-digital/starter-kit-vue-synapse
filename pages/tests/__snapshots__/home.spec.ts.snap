// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`home page should render > should render 1`] = `
"<div>
  <h1 class="text-primary">Accueil</h1>
  <div class="mt-6 mb-12 d-flex align-center justify-center">
    <div class="v-card v-theme--light v-card--density-default v-card--variant-tonal d-flex" style="width: 695px;">
      <!---->
      <div class="v-card__loader">
        <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
          <!---->
          <div class="v-progress-linear__background" style="opacity: NaN;"></div>
          <div class="v-progress-linear__buffer" style="opacity: NaN; width: 0%;"></div>
          <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
            <div class="v-progress-linear__indeterminate">
              <div class="v-progress-linear__indeterminate long"></div>
              <div class="v-progress-linear__indeterminate short"></div>
            </div>
          </transition-stub>
          <!---->
        </div>
      </div>
      <!---->
      <!----><img src="/starter-kit.svg" alt="Starter kit">
      <!---->
      <!----><span class="v-card__underlay"></span>
    </div>
  </div>
  <div class="v-row justify-center">
    <div class="v-col-md-6 v-col-12">
      <div class="v-card v-theme--light bg-grey-lighten-4 v-card--density-default v-card--variant-elevated mx-auto">
        <!---->
        <div class="v-card__loader">
          <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
            <!---->
            <div class="v-progress-linear__background" style="opacity: NaN;"></div>
            <div class="v-progress-linear__buffer" style="opacity: NaN; width: 0%;"></div>
            <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
              <div class="v-progress-linear__indeterminate">
                <div class="v-progress-linear__indeterminate long"></div>
                <div class="v-progress-linear__indeterminate short"></div>
              </div>
            </transition-stub>
            <!---->
          </div>
        </div>
        <!---->
        <!---->
        <div class="v-card-title">Exemple de composant Nuxt</div>
        <div class="v-card-item">
          <!---->
          <div class="v-card-item__content">
            <!---->
            <!---->
            <div msg="Bonjour">
              <h1 class="text-primary">Hello World!</h1>
            </div>
          </div>
          <!---->
        </div>
        <!---->
        <!----><span class="v-card__underlay"></span>
      </div>
    </div>
    <div class="v-col-md-6 v-col-12">
      <div class="v-card v-theme--light bg-grey-lighten-4 v-card--density-default v-card--variant-elevated mx-auto">
        <!---->
        <div class="v-card__loader">
          <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
            <!---->
            <div class="v-progress-linear__background" style="opacity: NaN;"></div>
            <div class="v-progress-linear__buffer" style="opacity: NaN; width: 0%;"></div>
            <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
              <div class="v-progress-linear__indeterminate">
                <div class="v-progress-linear__indeterminate long"></div>
                <div class="v-progress-linear__indeterminate short"></div>
              </div>
            </transition-stub>
            <!---->
          </div>
        </div>
        <!---->
        <!---->
        <div class="v-card-title">Exemple de composant DS</div>
        <div class="v-card-item">
          <!---->
          <div class="v-card-item__content">
            <!---->
            <!----><a data-v-b9c7d584="" class="v-btn v-theme--light text-primary v-btn--density-default v-btn--size-default v-btn--variant-text vd-back-btn text-none px-0 pr-1" outlined="false"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
              <!----><span class="v-btn__content" data-no-activator=""><i data-v-b9c7d584="" class="M20,11V13H8L13.5,18.5L12.08,19.92L4.16,12L12.08,4.08L13.5,5.5L8,11H20Z mdi v-icon notranslate v-theme--light v-icon--size-default text-primary mr-1" aria-hidden="true"></i>Retour</span>
              <!---->
              <!---->
            </a><br><button type="button" class="v-btn v-btn--elevated v-theme--light bg-secondary v-btn--density-default v-btn--size-default v-btn--variant-elevated"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
              <!----><span class="v-btn__content" data-no-activator="">Bouton secondaire</span>
              <!---->
              <!---->
            </button><br><br><a href="https://digital-design-system.netlify.app/demarrer/introduction/" target="_blank"> Voir la documentation des composants du Design System. </a>
          </div>
          <!---->
        </div>
        <!---->
        <!----><span class="v-card__underlay"></span>
      </div>
    </div>
  </div>
</div>"
`;
